File: ROUND_1/CPP/Optimal/Vault_security.cpp
#include <iostream>
using namespace std;

int main() {
    long long n;
    cin >> n;

    // Negative Vault IDs are never valid
    if (n < 0) {
        cout << "NO" << endl;
        return 0;
    }

    long long original = n;
    long long reversed = 0;

    while (n > 0) {
        reversed = reversed * 10 + n % 10;
        n /= 10;
    }

    if (reversed == original)
        cout << "YES" << endl;
    else
        cout << "NO" << endl;

    return 0;
}
/*
Time Complexity: O(log10 N)
Space Complexity: O(1)
*/

File: ROUND_1/JAVA/Optimal/Vault_security.java
import java.io.*;
import java.util.*;

public class Vault_security {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();

        // Negative numbers are never palindromes
        if (n < 0) {
            System.out.println("NO");
            return;
        }

        int original = n;
        int reverse = 0;

        while (n != 0) {

            int digit = n % 10;
            reverse = reverse * 10 + digit;
            n /= 10;
        }

        if (reverse == original)
            System.out.println("YES");
        else
            System.out.println("NO");
    }
    
}
/*
Time Complexity
O(log10 N)
Space Complexity
O(1)
*/
